core/models.py

```diff
@@ -4,3 +4,43 @@
 
 class User(AbstractUser):
     pass
+
+
+class Genre(models.Model):
+    genre_name = models.CharField(max_length=100)
+    genre_is_for_adults = models.BooleanField(default=False)
+
+
+class Actor(models.Model):
+    actor_name = models.CharField(max_length=100)
+    actor_age = models.PositiveIntegerField()
+    actor_nationality = models.CharField(max_length=100)
+
+
+class Movie(models.Model):
+    movie_title = models.CharField(max_length=100)
+    movie_genres = models.ManyToManyField(Genre)
+    movie_actors = models.ManyToManyField(Actor)
+    movie_duration = models.PositiveIntegerField()
+
+
+class CinemaHall(models.Model):
+    cinema_hall_name = models.CharField(max_length=100)
+
+
+class Seat(models.Model):
+    seat_row = models.PositiveIntegerField()
+    seat_number = models.PositiveIntegerField()
+    seat_hall = models.ForeignKey(CinemaHall, on_delete=models.PROTECT)
```

Comment:
I personally dony like PROTECT as it only encourages to delete more, not really protect, but it works fine I guess :D